models/inventory/StockTransfer.ts

```diff
@@ -373,6 +421,64 @@ export abstract class StockTransfer extends Transfer {
 
     return invoice;
   }
+
+  async getReturnDoc(): Promise<StockTransfer | null> {
+    if (!this.items?.length) {
+      return null;
+    }
+
+    const docData = this.getValidDict(true, true);
+    const docItems = docData.items as StockTransferItem[];
+    const returnDocItems: StockTransferItem[] = [];
+
+    const returnBalanceItemsQty = await this.fyo.db.getReturnBalanceItemsQty(
+      this.schema.name,
+      this.name!
+    );
+
+    for (const item of docItems) {
+      if (!item.quantity) {
+        continue;
+      }
+
+      let quantity = -1 * item.quantity;
+
+      if (returnBalanceItemsQty) {
+        const balanceItemQty = returnBalanceItemsQty.filter(
+          (i) => i.item === item.item
+        )[0];
+
+        if (!balanceItemQty) {
+          continue;
+        }
+        quantity = balanceItemQty.quantity as number;
```

Comment:
Better typing in the `getReturnBalanceItemsQty` would've caused you to not require assertions here.

Now, if someone is debugging an issue cause by balance quantities they'll have to trace back all the way to the query in `bespoke.ts` to figure out what the actual type is.

Reply:
corrected